src/lib/Repository/UiLocationService.php

```diff
@@ -52,14 +63,38 @@ public function loadLocations(ContentInfo $contentInfo)
         return $uiLocations;
     }
 
+    /**
+     * Deletes locations.
+     * Throws exception if there is an attempt to delete the main location.
+     *
+     * @param array $locationIds
+     * @param int $contentId
+     *
+     * @throws ForbiddenException
+     */
+    public function deleteLocations(array $locationIds, int $contentId)
+    {
+        $contentInfo = $this->contentService->loadContentInfo($contentId);
+
+        foreach ($locationIds as $locationId) {
+            $location = $this->locationService->loadLocation($locationId);
+
+            if ($this->isMainLocation($location, $contentInfo)) {
+                throw new ForbiddenException('Main location cannot be deleted.');
+            }
+
+            $this->locationService->deleteLocation($location);
```

Comment:
in addition the comment about disabling checkboxes, this can throw an UnauthorizedException and we might have the case where this exception is thrown for one Location but others can be/have been deleted. So I guess, there should a try catch there.